app/assets/javascripts/discourse/app/models/group.js

```diff
@@ -127,6 +127,12 @@ const Group = RestModel.extend({
     });
   },
 
+  join() {
+    return ajax(`/groups/${this.id}/join.json`, {
+      type: "PUT",
+    }).then(() => this.findMembers());
```

Comment:
Why do we need to call `findMembers` after the user has joined the group?

Reply:
Sadly, the code around of this place relies on side-effects of this method. We need to call it to get UI updated. And, by the way, I've reverted the commit that make it return the promise instead of swallowing it (see @eviltrout comment above) because it caused timeout when running client side tests.

Reply:
That's quite weird - we should add a TODO to fix that because we shouldn't be relying on swallowing a promise.

Reply:
Yep, I agree, I'm going to address all these methods that call `findMembers` altogether. Will create a todo.